### PR TITLE
Ignore non-X509 certificates in SignedCms

### PR DESCRIPTION
**Pull request by vcsjones**
Partial cherry-pick of the work @bartonjs [started](https://github.com/bartonjs/runtime/commit/ff6795eaf89bc5d68bb2e42a62a204b93240029c) and added a test.

The test uses `SignedCms` and not `Rfc3161TimestampToken` since the example TST that reproduces the issue has other, unrelated issued with being a valid TST.

Contributes to #62307.

### CODE REVIEW

**Comment by vcsjones** [comment]
Draft because I want to see how Windows handles this in CI first.

**Comment by vcsjones** [comment]
Looks like everything "passed" on Windows, and then infrastructure failed to report results. " END OF WORK ITEM LOG: Command timed out, and was killed]".

File: src/libraries/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs [hunk 1]
```diff
@@ -1472,5 +1472,159 @@ internal static class SignedDocuments
             "2B78166BDB1612848B535ADC3F0E7BE52991A17F48AFDCCC1698A236BA338930" +
             "50EBAAC4460DAA35185C16670F597E0E6E0CB0AA83F51AAEF452F3367DD9350A" +
             "8A49A5A8F79DF8E921303AB5D6646A482F0F59D9980310E1AE3EE8D77CB857").HexToByteArray();
+
+        internal static readonly byte[] TstWithAttributeCertificate = (
+            "308212E306092A864886F70D010702A08212D4308212D0020103310F300D0609" +
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs:1477]: Your blob is a lot smaller than mine was.  Where'd it come from? :smile:

Comment (vcsjones) [reply src/libraries/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs:1477]: Your blob was an Authenticode signature with a counter signature. This blob is just the counter signature (RFC3161 timestamp).

**Review by bartonjs — APPROVED** [review]

**Pull request merged**